tests/NCronJob.Tests/RunDependentJobTests.cs

```diff
@@ -36,6 +36,35 @@ public async Task WhenJobWasFailed_DependentJobShouldRun()
         result.ShouldBe("Me: Message Parent: Failed");
     }
 
+    [Fact]
+    public async Task RemovingAJobShouldAlsoRemoveItsDependencies()
+    {
+        ServiceCollection.AddNCronJob(n => n.AddJob<MainJob>()
+            .ExecuteWhen(success: s => s.RunJob<SubMainJob>()));
+        var provider = CreateServiceProvider();
+        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+
+        var instantJobRegistry = provider.GetRequiredService<IInstantJobRegistry>();
+        instantJobRegistry.ForceRunInstantJob<MainJob>();
+
+        var result = await CommunicationChannel.Reader.ReadAsync(CancellationToken);
+        Assert.Equal(nameof(MainJob), result);
+        result = await CommunicationChannel.Reader.ReadAsync(CancellationToken);
+        Assert.Equal(nameof(SubMainJob), result);
+
+        var registry = provider.GetRequiredService<IRuntimeJobRegistry>();
+
+        registry.RemoveJob<PrincipalCorrelationIdJob>();
+        registry.AddJob(n => n.AddJob<MainJob>());
```

Comment:
@linkdotnet FWIW, this syntax (`AddJob(AddJob)`) seems a bit weird.

Wouldn't there be, in the context of v4, an opportunity to find a less "redundant" naming?

Reply:
Yes - what I wanted is to reuse the same builder so that people only have to "learn" one API.

Reply:
Tracked through #122